File: ant-sdk/antd-zig/examples/02-data.zig
const std = @import("std");
const antd = @import("antd");

pub fn main() !void {
    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
    defer _ = gpa.deinit();
    const allocator = gpa.allocator();

    var client = antd.Client.init(allocator, antd.default_base_url);
    defer client.deinit();

    // Store public data
    const message = "Hello, Autonomi!";
    std.debug.print("Storing public data: {s}\n", .{message});

    const put_result = try client.dataPutPublic(message, .auto);
    defer put_result.deinit(allocator);

    std.debug.print("Stored at address: {s}\n", .{put_result.address});
    std.debug.print("Chunks stored: {d}, payment mode: {s}\n", .{ put_result.chunks_stored, put_result.payment_mode_used });

    // Retrieve public data
    const data = try client.dataGetPublic(put_result.address);
    defer allocator.free(data);

    std.debug.print("Retrieved: {s}\n", .{data});

    // Estimate storage cost
    const est = try client.dataCost("some data to estimate", .auto);
    defer est.deinit(allocator);

    std.debug.print(
        "Estimate: {d} bytes in {d} chunks, storage {s} atto, gas {s} wei, mode {s}\n",
        .{ est.file_size, est.chunk_count, est.cost, est.estimated_gas_cost_wei, est.payment_mode },
    );
}

File: ant-sdk/antd-rust/examples/02-data.rs
use antd_client::{Client, PaymentMode, DEFAULT_BASE_URL};

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let client = Client::new(DEFAULT_BASE_URL);

    // Store public data
    let result = client
        .data_put_public(b"Hello, Autonomi!", PaymentMode::Auto)
        .await?;
    println!(
        "Stored at: {} (chunks: {}, mode: {})",
        result.address, result.chunks_stored, result.payment_mode_used
    );

    // Retrieve public data
    let data = client.data_get_public(&result.address).await?;
    println!("Retrieved: {}", String::from_utf8_lossy(&data));

    // Estimate cost before storing
    let est = client
        .data_cost(b"Some data to estimate", PaymentMode::Auto)
        .await?;
    println!(
        "Estimate: {} bytes in {} chunks, storage {} atto, gas {} wei, mode {}",
        est.file_size, est.chunk_count, est.cost, est.estimated_gas_cost_wei, est.payment_mode
    );

    Ok(())
}
